python/pyarrow/pandas_compat.py

```diff
@@ -404,7 +404,7 @@ def table_to_blockmanager(options, table, memory_pool, nthreads=1):
             index_name = None if is_unnamed_index_level(name) else name
             col_pandas = col.to_pandas()
             values = col_pandas.values
-            if not values.flags.writeable:
+            if hasattr(values, 'flags') and not values.flags.writeable:
```

Comment:
For categorical columns, we need to check `values.categories.values` and `values.codes` for writability. You can check for categorical columns either by using the dtype or `isinstance(value, pd.Categorical)`

Reply:
@xhochy Do we need to check writability of `Categorical.codes` and `Categorical.categories .values` even if we cannot edit directly?
```
>>> import pandas as pd
>>> import numpy as np
>>> values = pd.Categorical(['A', 'A'])
>>> 
>>> arr = pa.DictionaryArray.from_arrays(
...     np.array([0, 0]),
...     np.array([100]))
>>>
>>> result = arr.to_pandas()
>>>
>>> result.categories.values
array([100])
>>> result.categories.values[:] = 1
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: assignment destination is read-only
>>> 
>>> result.codes
array([0, 0], dtype=int8)
>>> result.codes[:] = 1
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: assignment destination is read-only
```

Reply:
@Licht-T if they pose a problem with Pandas 0.19.2 and we continue to support this version, we should check them. If we would switch our requirements to `pandas>=0.20`, then we could ignore them. Maybe @wesm or @cpcloud has some opinion on dropping older Pandas versions.

Reply:
Thanks @xhochy, I got it. Should I leave a note about the problem on code?

Reply:
I think this check here is a workaround for a pandas issue (where `pd.factorize` will fail on non-writeable arrays because of stuff with the Cython buffer protocol). I don't think this will impact Categorical types, @jreback can you confirm?